plugins/HostLinux.cpp

```diff
@@ -32,112 +32,127 @@ bool HostLinux::peek(const procptr_t address, void *dst, const size_t size) cons
 	return (ret != -1 && static_cast< size_t >(ret) == in.iov_len);
 }
 
-procptr_t HostLinux::module(const std::string &module) const {
-	std::stringstream ss;
-	ss << "/proc/";
-	ss << m_pid;
-	ss << "/maps";
-	const auto maps = readFile(ss.str());
+Modules HostLinux::modules() const {
+	std::ostringstream path;
+	path << "/proc/";
+	path << m_pid;
+	path << "/maps";
+	const auto maps = readFile(path.str());
 
 	if (maps.size() == 0) {
-		return 0;
+		return {};
 	}
 
-	std::stringstream ssPath(maps);
-	while (ssPath.good()) {
-		std::string baseaddr;
-
-		int ch;
-		while (1) {
-			ch = ssPath.get();
-			if (ch == '-') {
-				break;
-			} else if (ch == EOF) {
-				return 0;
+	Modules modules;
+	std::string name;
+	std::stringstream stream(maps);
+
+	while (stream.good()) {
+		std::string buf;
+
+		for (auto ch = stream.get(); ch != '-'; ch = stream.get()) {
+			if (stream.eof()) {
+				return modules;
```

Comment:
Don't you want to process the current contents of `buf` before returning `modules`?

Reply:
No, we cannot trust the data we read if the stream unexpectedly finishes.

Reply:
ok